judge/utils/mail.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+from django.conf import settings
+from django.core.mail import EmailMultiAlternatives
+from django.forms import ValidationError
+from django.template import loader
+from django.utils.translation import gettext
+
+
+bad_mail_regex = list(map(re.compile, settings.BAD_MAIL_PROVIDER_REGEX))
+
+
+def validate_email_domain(email):
+    if '@' in email:
+        domain = email.split('@')[-1].lower()
+        if domain in settings.BAD_MAIL_PROVIDERS or any(regex.match(domain) for regex in bad_mail_regex):
+            raise ValidationError(gettext('Your email provider is not allowed due to history of abuse. '
+                                          'Please use a reputable email provider.'))
+
+
+def send_mail(
+    subject_template_name,
+    email_template_name,
```

Comment:
The ordering of these parameters is a little strange, it'd be good to keep everything template-related together, not separated by email parameters.

Reply:
This was originally copied from https://github.com/django/django/blob/main/django/contrib/auth/forms.py#L287. I think their parameters were ordered that way since the HTML template was optional. Since we want likely want our emails to always contain an HTML version, I will reorder the parameters.